From ae8c971899ac9090d496245843d51d8615f700f5 Mon Sep 17 00:00:00 2001
From: Stefan Berger <stefanb@linux.ibm.com>
Date: Wed, 2 Feb 2022 15:40:41 -0500
Subject: [PATCH 64/89] wip: ima: Restrict check on ima_fail_unverifiable_sigs
 to init_ima_ns

Restrict the check on ima_fail_unverifiable_sigs to the init_ima_ns.

Signed-off-by: Stefan Berger <stefanb@linux.ibm.com>
---
 security/integrity/ima/ima_policy.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/security/integrity/ima/ima_policy.c b/security/integrity/ima/ima_policy.c
index 7c86185c5d11..1ba1a84fa9c9 100644
--- a/security/integrity/ima/ima_policy.c
+++ b/security/integrity/ima/ima_policy.c
@@ -779,7 +779,7 @@ int ima_match_policy(struct ima_namespace *ns,
 		if (entry->action & IMA_APPRAISE) {
 			action |= get_subaction(entry, func);
 			action &= ~IMA_HASH;
-			if (ima_fail_unverifiable_sigs)
+			if (ns == &init_ima_ns && ima_fail_unverifiable_sigs)
 				action |= IMA_FAIL_UNVERIFIABLE_SIGS;
 
 			if (allowed_algos &&
-- 
2.37.3

